### PR TITLE
Optimize `SetProperty` for inline styles

### PR DESCRIPTION
**Pull request by pcwalton**
r? @Ms2ger

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6823)

### CODE REVIEW

**Comment by pcwalton** [comment]
also r? @SimonSapin for the style pieces

**Comment by SimonSapin** [comment]
I have some of this already in https://github.com/servo/servo/pull/6741

**Comment by metajack** [comment]
@SimonSapin can we get yours landed then so this can be rebased?

**Comment by SimonSapin** [comment]
I’m in the middle of addressing review comments on #6741.

File: components/script/dom/element.rs [hunk 1]
```diff
@@ -89,6 +89,7 @@ use std::ascii::AsciiExt;
 use std::borrow::{Cow, ToOwned};
 use std::cell::{Ref, RefMut};
 use std::default::Default;
+use std::intrinsics;
```
Comment (Ms2ger) [inline components/script/dom/element.rs]: Unused?

**Comment by pcwalton** [comment]
Rebased.

r? @Ms2ger

File: components/style/properties.mako.rs [hunk 2]
```diff
@@ -5679,8 +5679,8 @@ pub fn parse_one_declaration(name: &str, input: &str, base_url: &Url)
     }
 }
 
-struct PropertyDeclarationParser<'a, 'b: 'a> {
-    context: &'a ParserContext<'b>,
+pub struct PropertyDeclarationParser<'a, 'b: 'a> {
+    pub context: &'a ParserContext<'b>,
```
Comment (SimonSapin) [inline components/style/properties.mako.rs]: Why this change? It doesn’t seem to be used.

File: components/style/properties.mako.rs [hunk 3]
```diff
@@ -5838,6 +5838,15 @@ pub enum PropertyDeclaration {
     Custom(::custom_properties::Name, DeclaredValue<::custom_properties::SpecifiedValue>),
 }
 
+impl PropertyDeclaration {
+    #[inline]
+    pub fn id(&self) -> u64 {
+        unsafe {
+            intrinsics::discriminant_value(self)
```
Comment (SimonSapin) [inline components/style/properties.mako.rs]: This uses the same ID for all custom properties. Maybe return `Atom` instead? (With static atoms for non-custom properties)

**Comment by Ms2ger** [comment]
r- 0dcb42a21956f3712f8770ea6edaaaa974310dd9; it doesn't do anything.
r+ 1bd179edc1d143cd165c8b1fd8aa04ac04f1bca7 if you revert the changes to `fallback_base_url`; they will need to be reverted anyway when we implement the `about:blank`/`srcdoc` parts.
r? SimonSapin 1699fe87493c718ae3729d9f6b40da6eb6d6e2e7 (if you address [my earlier comment about the import](https://github.com/servo/servo/pull/6823#discussion_r35860632))
r+ 668a668ddbc6556a7050621f7b0ca41b2adbfd11

File: components/style/properties.mako.rs (hunk 2 above)
Comment (SimonSapin) [inline components/style/properties.mako.rs]: Ah, it’s probably left over from a rebase. As Ms2ger says, the commit can be skipped.

**Comment by pcwalton** [comment]
Split the parts that have r+ into https://github.com/servo/servo/pull/7736

**Comment by pcwalton** [comment]
Going to go ahead and close this since the important stuff has landed.
